File: apple/Sources/CCDAEngine/Models/Media/CCDAMediaRepresentation.swift
import Foundation

/// C-CDA media representation metadata.
public enum CCDAMediaRepresentation: Hashable, Sendable {
    case base64
    case text
    case unsupported(String)
    case unknown

    /// Creates a representation from the XML value.
    public init(rawValue: String?) {
        guard let rawValue, !rawValue.isEmpty else {
            self = .unknown
            return
        }

        switch rawValue.uppercased() {
        case "B64":
            self = .base64
        case "TXT":
            self = .text
        default:
            self = .unsupported(rawValue)
        }
    }

    /// XML representation value when known.
    public var rawValue: String? {
        switch self {
        case .base64:
            return "B64"
        case .text:
            return "TXT"
        case .unsupported(let value):
            return value
        case .unknown:
            return nil
        }
    }
}

File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Media/CCDAMediaRepresentation.kt
package com.ccdakit.engine

/** C-CDA media representation metadata. */
sealed class CCDAMediaRepresentation(open val rawValue: String?) {
    /** Base64-encoded payload representation. */
    data object Base64 : CCDAMediaRepresentation("B64")
    /** Text payload representation. */
    data object Text : CCDAMediaRepresentation("TXT")
    /** Unsupported representation preserved from source XML. */
    data class Unsupported(override val rawValue: String) : CCDAMediaRepresentation(rawValue)
    /** Missing or empty representation. */
    data object Unknown : CCDAMediaRepresentation(null)

    companion object {
        /** Creates a representation from the XML value. */
        fun fromRawValue(rawValue: String?): CCDAMediaRepresentation = when (rawValue?.uppercase()) {
            null, "" -> Unknown
            "B64" -> Base64
            "TXT" -> Text
            else -> Unsupported(rawValue)
        }
    }
}
